src/gui/spatialIndicator.js

```diff
@@ -0,0 +1,370 @@
+    `;
+    
+    const cylinderFragmentShader = `
+        #define S(a, b, x) smoothstep(a, b, x)
+        #define PI 3.14159
+        // #define blur 0.002
+        #define blur 0.1
+        #define brightness 0.001
+        
+        #define black vec3(0.)
+        #define white vec3(1.)
+        #define red vec3(1., 0., 0.)
+        #define green vec3(0., 1., 0.)
+        #define blue vec3(0., 0., 1.)
+        #define cyan vec3(0., 1., 1.)
+        
+        struct Lines {
+            float width;
+            float height;
+            float x;
+            float y;
+            float speed;
+        };
+        
+        uniform int amount;
+        // cannot initialize the lines[] array with variable size
+        // so update the amount in the js file and shader always the same
+        uniform Lines lines[5];
+        
+        // set up color uniforms
+        struct AvatarColor {
+            vec3 color;
+            vec3 colorLighter;
+        };
+        uniform AvatarColor avatarColor[1];
+        
+        varying vec2 vUv;
+        
+        // draw a vertical line segment at p with width w and height h
+        float Line(vec2 uv, vec2 p, float w, float h) {
+            uv -= p;
+            // un-comment below line to find out what I did wrong
+            // float horizontal = S(.01, 0., abs(length(uv.x - w / 2.)));
+            float horizontal = S(blur, 0., abs(uv.x)- w / 2.);
+            float vertical = S(blur, 0., abs(uv.y) - h / 2.);
+            return horizontal * vertical;
+        }
+        
+        float GlowingLine(vec2 uv, vec2 p, float w, float h) {
+            uv -= p;
+            float horizontal = S(blur, 0., abs(uv.x)- w / 2.);
+            float vertical = S(blur, 0., abs(uv.y) - h / 2.);
+            float d = horizontal * vertical;
+            float fx = brightness / abs(d - 1.);
+            fx = pow(fx, .5);
+            return fx;
+        }
+        
+        void main(void) {
+            vec2 uv = vUv;
+        
+            vec3 col = vec3(0.);
+            float alpha = 0.;
+            
+            // draw the ascending lines
+            for (int i = 0; i < amount; i++) {
+                float x = lines[i].x, y = lines[i].y, width = lines[i].width, height = lines[i].height;
+                float d = GlowingLine(uv, vec2(x, y), width, height);
+                col += avatarColor[0].color * d;
+                alpha += d;
+            }
+        
+            col *= .1;
+            alpha *= .1;
+            if (alpha < .35) alpha = 0.;
+        
+            // draw the fluctuating upper alpha fade out boundary
+            // float boundary = S(0.0, 0.8, 1. - uv.y);
+            // float boundary = S(0.8, 0.0, 1. - uv.y);
+            // col = mix(col, red, boundary);
+        
+            // float boundary = S(0.2, 0.0, 1. - uv.y);
+            // alpha = mix(alpha, 0., boundary);
+        
+            gl_FragColor = vec4(col, alpha);
+        }
+    `;
+
+    const amount = 5;
```

Comment:
Could use a comment here corresponding to the one above mentioning how they need to be kept in sync